registry/base_registry.go

```diff
@@ -213,24 +268,31 @@ func (r *BaseRegistry) register(c common.URL) error {
 	switch role {
 
 	case common.PROVIDER:
-		dubboPath, rawURL, err = r.providerRegistry(c, params)
+		dubboPath, rawURL, err = r.providerRegistry(c, params, createPathFunc)
 	case common.CONSUMER:
-		dubboPath, rawURL, err = r.consumerRegistry(c, params)
+		dubboPath, rawURL, err = r.consumerRegistry(c, params, createPathFunc)
 	default:
 		return perrors.Errorf("@c{%v} type is not referencer or provider", c)
 	}
 	encodedURL = url.QueryEscape(rawURL)
 	dubboPath = strings.ReplaceAll(dubboPath, "$", "%24")
-	err = r.facadeBasedRegistry.DoRegister(dubboPath, encodedURL)
+	err = f(dubboPath, encodedURL)
 
 	if err != nil {
 		return perrors.WithMessagef(err, "register Node(path:%s, url:%s)", dubboPath, rawURL)
 	}
 	return nil
 }
 
+// createPath will create dubbo path in register
+func (r *BaseRegistry) createPath(dubboPath string) error {
+	r.cltLock.Lock()
+	defer r.cltLock.Unlock()
+	return r.facadeBasedRegistry.CreatePath(dubboPath)
```

Comment:
have u got the reason of lock facadeBasedRegistry.CreatePath by r.cltLock.Lock()?

Reply:
the source code as below, i just extract it.
https://github.com/apache/dubbo-go/blob/8a3c02c9b7737a12712451b16a3cdc8ce46a5c7d/registry/base_registry.go#L240

https://github.com/apache/dubbo-go/blob/8a3c02c9b7737a12712451b16a3cdc8ce46a5c7d/registry/base_registry.go#L288

https://github.com/apache/dubbo-go/blob/8a3c02c9b7737a12712451b16a3cdc8ce46a5c7d/registry/base_registry.go#L300